packages/vit-s/src/generate-alt-text.js

```diff
@@ -0,0 +1,24 @@
+/**
+   * Method for converting an array with strings to a natural sentence,
+   * concatenating the items with commans, and the last item with 'and'.
+   * @param {Array} arrayWithTextItems Array with text items.
+   * @returns {string} Natural language sentence in the form of 'a, b and c', where a/b/c are items.
+   */
+function arrayToString(arr) {
+  // arrayWithTextItems = arrayWithTextItems.filter(item => item !== undefined);
+  if (arr.length === 0) {
+    return '';
+  }
+  if (arr.length === 1) {
+    return arr[0];
+  }
+  return ''.concat(arr.slice(0, -1).join(', '), ' and ', arr.slice(-1));
+}
+
+const viewTypesToText = new Map([['scatterplot', 'scatterplot with embeddings'], ['gating', 'scatterplot of gated gene expression data'], ['heatmap', 'cell by gene heatmap'], ['spatial', 'spatial view'], ['area', 'area displayed'], ['layerController', 'layer controller'], ['genomicProfiles', 'genome browser tracks with bar plots'], ['featureList', 'interactive list of features'], ['obsSets', 'list of potential observation sets'], ['obsSetSizes', 'sizes of selected observation sets'], ['description', 'description'], ['status', 'status view for debugging'], ['obsSetFeatureValueDistribution', 'violin plot with values'], ['featureValueHistogram', 'distribution of values']]);
```

Comment:
Good plan! Though there still needs to be some mapping from the viewtype to natural language, instead of constants_values -> natural_language, I now made the mapping from constants -> natural_language, and then retrieve the mapping of constants -> constants_values and merge them together.

Reply:
Yes you are correct, I meant that

```js
const viewTypesToText = new Map([['scatterplot', 'scatterplot with embeddings'], ['gating', 'scatterplot of gated gene expression data'],
```

could become

```js
const viewTypesToText = new Map([[ViewType.SCATTERPLOT, 'scatterplot with embeddings'], [ViewType.GATING, 'scatterplot of gated gene expression data'],
```

This results in the same `Map` keys as you had originally, and just helps a bit to avoid accidental typos in the keys. It should avoid the need for the `getViewTypesMap` function